### ISSUE TITLE
(iOS) Frida : Failed to attach: the connection is closed

### ISSUE THREAD

**Issue by masbog** [frida/frida#358]
Hello all,

Phone : i used frida-server ran on iOS 9.3.2 iPhone 5s with version of frida from Cydia 10.6.10.
Mac : i compile from this source, and version is 10.6.9

$ frida-ps -U -> OK, i getting PID and Application name

but i try attach using frida some process PID from mac got error "Failed to attach: the connection is closed"

example: 
masbog$ frida -U -l SSLKillSwitch.js -p 3924 --debug
     ____
    / _  |   Frida 10.6.9 - A world-class dynamic instrumentation framework
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at http://www.frida.re/docs/home/
Failed to attach: the connection is closed
masbog$

can someone help me fix this problem?

thank you all 👍

**Comment by csftech** [comment]
@masbog Try the latest frida.

**Comment by cpelliccioni** [comment]
I have installed the latest frida (10.6.54) on a iOS 11.1.2 but I see the same message.

**Comment by csftech** [comment]
@cpelliccioni Frida doesn't support spawning on iOS 11 now. I only tested on iPhone 6S 11.1.2 jailbreaked with Electra Beta 11-3.
Frida should work normally on iOS 8 to iOS 10.

**Comment by galenbwill** [comment]
Also confirmed on iPhone 6 11.1.2 with latest Electra release 1.0.2, installing frida from cydia repo.

(Also tried building from master with same result.)

But as you said...
> Frida doesn't support spawning on iOS 11 now.

**Comment by oleavr** [comment]
> Frida doesn't support spawning on iOS 11 now.

The issue is that some (all?) of the new iOS 11 jailbreaks interfere with how apps are launched, which is also something Frida has been doing all along by injecting code into [launchd](https://github.com/frida/frida-core/blob/b46081304f904773882f6efac4507c627a38fcdb/src/darwin/agent/launchd.js) and [xpcproxy](https://github.com/frida/frida-core/blob/b46081304f904773882f6efac4507c627a38fcdb/src/darwin/agent/xpcproxy.js). Previous jailbreaks also tampered with the app launching logic, but quite differently from the current ones. So now we have the jailbreak and Frida stepping on each others' toes during the early instrumentation.

(I didn't investigate this myself yet, kudos to @mrmacete for digging into this. Investigation isn't yet complete, though, so these are just some early observations.)

**Comment by masbog** [comment]
awh... just add "--no-pause" option fix for connection closed , can it add to default option just for iOS? seems got trouble pause after attach by package id only 🤗

**Comment by oleavr** [comment]
@masbog Yeah if you don't type `%resume` fast enough, the system will time out the launch and kill the process. Changing that behavior isn't something we can do without bumping the major version – it would be a breaking change. Instead I would suggest contributing an improvement to Frida that disables the timeout by hooking SpringBoard... :)

**Comment by bensh** [comment]
also getting the same failed to attach issue even with the --no-pause switch

Frida 10.6.21
iOS 9.0.2

**Comment by oleavr** [comment]
@bensh Please don’t report bugs unless you’re running the latest version of Frida — you might be reporting issues that have already been fixed.

**Comment by bensh** [comment]
Apologies, I wrote that comment then went to update in case it had been fixed then forgot to change the version. I am running 10.7.7

**Comment by vladimirarevshatyan** [comment]
Same happens on Android rooted devices, latest FRIDA server at this moment is pushed, no clue what's wrong, and same goes with --no-pause
